worker/uniter/uniter.go

```diff
@@ -236,6 +236,9 @@ func (u *Uniter) init(unitTag names.UnitTag) (err error) {
 	u.storage = storageAttachments
 	u.addCleanup(storageAttachments.Stop)
 
+	if err := charm.ClearDownloads(u.paths.State.BundlesDir); err != nil {
+		return errors.Annotatef(err, "cannot clear downloads")
```

Comment:
This probably shouldn't be fatal should it?... Might be best to just warn instead.